### PR TITLE
avatar support

### LINKED ISSUE #2451: Add Avatar Support in Thread List and Details View
I recently implemented avatars in the thread list and details view on our iOS app.
Used Letter Avatars for the time being(Couldn't find a way to show actual avatars), but it definitely spruces up the UI.

Here's what it looks like on iOS:
image

It'd be awesome to mirror this feature on the Android side. What do you think?

Ref: https://github.com/FlowCrypt/flowcrypt-ios/pull/2320

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added avatars to the messages list.

close #2451

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)
--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

**Review by ioanmo226 — COMMENTED** [review]
Quick code review. Will do deep review after my comment is addressed.
How about adding avatar in thread detail page as well like iOS app? (also take care of draft icon in thread detail page in iOS side)
image

Also Avatar circle seem a bit large and it contains almost 1/5 of screen.
Maybe better if we make a bit smaller?
Screenshot_3

File: FlowCrypt/src/main/java/com/flowcrypt/email/extensions/android/widget/ImageViewExt.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        ).apply(RequestOptions().transform(MultiTransformation(CenterCrop(), CircleCrop())))
+    } else {
+      requestBuilder
+        .apply(
+          RequestOptions().transform(CenterCrop()).placeholder(placeholderId).error(placeholderId)
+        ).apply(RequestOptions().transform(CenterCrop()))
+    }
+  }
+
+  requestBuilder.into(this)
```
[the remark this answers is not in the document]
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/extensions/android/widget/ImageViewExt.kt:76]: > I think we won't need castedSource as Glide can handle multiple types well.

Ha... you are right. This code was copied from an old source. Looks like `Glide` changed the code some time ago to be able to use `Any`. Good catch!

> Also I think we can refactor code like below

Perfect :+1:

**Comment by DenBond7** [comment]
> How about adding avatar in thread detail page as well like iOS app? (also take care of draft icon in thread detail page in iOS side)

Sure. Will do

> Also Avatar circle seem a bit large and it contains almost 1/5 of screen.
Maybe better if we make a bit smaller?

Could you share the screen resolution for iOS and Android devices?

**Comment by ioanmo226** [comment]
iOS one.
image

**Comment by DenBond7** [comment]
I meant real screen resolution like 1920*1080 and the screen size in cm(or inches)

**Comment by DenBond7** [comment]
What about this one? `48dp` instead of `56dp`

| 56dp  | 48dp |
| ------------- | ------------- |
| image  | image  |

**Comment by ioanmo226** [comment]
I'm using `iPhone 14 Plus` simulator and Pixel 3a simulator.
Looks much better

**Comment by ioanmo226** [comment]
@DenBond7 Isn't draft displayed in thread detail page?
I can see draft in `Drafts` folder but can't see in thread detail page.

**Comment by DenBond7** [comment]
Android doesn't support Gmail threads yet. It uses `a single message` logic. Because we started development from the IMAP implementation.

**Comment by ioanmo226** [comment]
OHh, Yep!
I missed it. Sorry

**Review by ioanmo226 — APPROVED** [review]
👍

**Pull request merged**